src/sentry/taskworker/worker.py

```diff
@@ -226,6 +194,23 @@ def handle_alarm(signum: int, frame: FrameType | None) -> None:
             with timeout_alarm(activation.processing_deadline_duration, handle_alarm):
```

Comment:
What if we wrapped this and the sigalarm in a function?

```
func execute(...) -> bool:
    did_timeout = False
    def handle_alarm(...) -> None:
        did_timeout = True

    try:
        with timeout_alarm(activation..., handle_alarm):
            _execute_activation(...)
    finally:
        return did_timeout
```

This would get around the possibility of code catching `BaseException`?

Reply:
That could work well. We'd need to keep the existing capture_exception() logic as well 🤔